properties/pkg/property.go

```diff
@@ -0,0 +1,178 @@
+// Copyright © 2021 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package properties
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+const (
+	ListSeparator = ","
+)
+
+type PropertyType uint
+
+const (
+	Int PropertyType = iota
+	Float
+	Bool
+	String
+	List
+	Invalid
+)
+
+// Property is used to store the key and value for a property. It also provides methods
+// to cast the value of the property into different types.
+type Property struct {
+	// key holds the name of the property in string.
+	key string
+	// value stores the value of the property in string.
+	value string
+	// comment stores comments for the property
+	comment string
+}
+
+// Key method return the name of the Property.
+func (p *Property) Key() string {
```

Comment:
Yeah. The idea here is to keep the `Property` object readonly. Although I have no strong argument against making its fields exported, I would just keep it as is for now. We can change this when we have use cases where it would be more convenient to make the `Property` mutable.